pallets/payment/src/lib.rs

```diff
@@ -415,43 +497,66 @@ pub mod pallet {
 					}
 					// Calculate incentive amount - this is to insentivise the user to release
 					// the funds once a transaction has been completed
-					let incentive_amount = T::IncentivePercentage::get().mul_floor(amount);
+					let incentive_amount = match charge_incentive_amount {
+						true => T::IncentivePercentage::get().mul_floor(amount),
```

Comment:
I think I'd still prefer to have this incentive be passed explicitly as a parameter and calculate the amount if the percentage passed is not 0. I'm not a big fan of a boolean as function parameters as they make code harder to read and less maintainable since when you see `some_function(true)` you don't really have context and have to dig deeper to see what it actually means.